### PR TITLE
Recipe app

### PR DESCRIPTION
**Pull request by husush23**
# We are displaying the recipe app for the review in this PR , in the app functionality we have so much things.
### Here are some pages of the app:
- Root/public recipes
- Foods
- Recipes
- Recipe's details
- Shopping list
Apart from those the app has a bunch of functionalities, the user can sign up/in, it also confirms the user's identity via an email confirmation link. Additionally, the user has only limited access to available public recipes, he/she can only see if the recipe is not theirs, or it can be managed by them.

### Finally is well-designed and added both unit and integration tests for proper functioning.
We are waiting to hear about your review👂🏻👂🏻😊

### CODE REVIEW

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ General shopping list generated and working perfectly 📌 
✔️ Added tests 📌 
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Your name repeats twice in the Authors section of your README. Please ensure to include the actual name of your coding partner 🙏 
image
- [ ] The link to your License file is currently broken. Please fix it as it is essential that developers are informed on how to legally use your code 👍 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/public_recipes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class PublicRecipesController < ApplicationController
+  def index
+    @public_recipes = Recipe.includes(:user, recipe_foods: :food).where(public: true).order(created_at: :desc)
+  end
```
[the remark this answers is not in the document]
Comment (husush23) [reply app/controllers/public_recipes_controller.rb:4]: Hey @codepantha  the public recipe link is available on the root file, you can click on the app name and you will. Please check the project well before block me to pass, make sure what are recommending us to do.

[the remark this answers is not in the document]
Comment (husush23) [reply app/controllers/public_recipes_controller.rb:4]: @codepantha What's the issue between if I use 'shopping list' or 'General shopping list ' that looks illogical thing to reject for this project, however I updated it as you requested.

File: README.md [hunk 2]
```diff
@@ -1 +1,195 @@
...
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I would like to express my heartfelt appreciation to Microverse for helping doing such projects, it's building our mindS and I am thankfully appreciating their efforts for heliping doing these tasks.
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (t-yanick) [inline README.md:193]: - [ ] As requested by the previous reviewer, kindly fix the MIT license link as it is still broken.
Screen Shot 2023-09-15 at 1 58 59 PM

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @husush23 and @Tshobohwa , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**